tools/integration_tests/emulator_tests/util/test_helper.go

```
@@ -16,78 +16,80 @@ package emulator_tests
 
 import (
 	"crypto/rand"
+	"errors"
 	"fmt"
 	"io"
 	"log"
 	"os"
 	"os/exec"
-	"path"
+	"regexp"
 	"strconv"
 	"strings"
 	"syscall"
 	"testing"
 	"time"
 
 	"github.com/googlecloudplatform/gcsfuse/v2/tools/integration_tests/util/operations"
-	"github.com/googlecloudplatform/gcsfuse/v2/tools/integration_tests/util/setup"
 )
 
-// StartProxyServer starts a proxy server as a background process and handles its lifecycle.
+const PortAndProxyProcessIdInfoRegex = `Listening Proxy Server On Port \[(\d+)\] with Process ID \[(\d+)\]`
+
+// StartProxyServer starts the proxy server in the background and returns the port and process ID it is listening on.
+//
+// It takes the config path and log file path as input. The proxy server is starts in background using
+// the go run command. The output of the proxy server is redirected to the log file. It also sets the
+// STORAGE_EMULATOR_HOST to `localhost:port` so that any new storage client will use this endpoint by
+// default. If any error occurs it fails.
 //
-// It launches the proxy server with the specified configuration and port, logs its output to a file.
-func StartProxyServer(configPath string) {
-	// Start the proxy in the background
-	cmd := exec.Command("go", "run", "../proxy_server/.", "--config-path="+configPath)
-	logFileForProxyServer, err := os.Create(path.Join(os.Getenv("KOKORO_ARTIFACTS_DIR"), "proxy-"+setup.GenerateRandomString(5)))
+// Parameters:
+//   - configPath: Path for config for Proxy Server.
+//   - logFilePath: Path for log file for Proxy Server Logs.
+//
+// Returns:
+//   - int: Port number that the proxy server is listening on.
+//   - int: Proxy Server Process ID.
+func StartProxyServer(configPath, logFilePath string) (int, int) {
```

Can we also return an error, and corresponding handling in the tests?

---

Done!